azkaban-webserver/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -518,6 +520,51 @@ private void handleDownloadProject(HttpServletRequest req,
 
   }
 
+    /**
+     * validate readiness of a project and user permission and use
+     * projectManager to purge the project if things looks good
+     **/
+    private void handlePurgeProject(HttpServletRequest req,
+        HttpServletResponse resp, Session session) throws ServletException,
+        IOException {
+        User user = session.getUser();
+        String projectName = getParam(req, "project");
+        HashMap<String, Object> ret = new HashMap<String, Object>();
+        boolean valid = true;
```

Comment:
Just thinking, should the variable "valid" be renamed to something better?
Does it stand for "shouldPurge", or does it stand for "purgeOperationSuccessful"?